applications/PoromechanicsApplication/custom_constitutive/simplified_bilinear_3D_law.hpp

```diff
@@ -0,0 +1,134 @@
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Javier San Mauro Saiz
+//                   Joaquin Irazabal Gonzalez
+//
+
+#if !defined (KRATOS_SIMPLIFIED_BILINEAR_3D_LAW_H_INCLUDED)
+#define  KRATOS_SIMPLIFIED_BILINEAR_3D_LAW_H_INCLUDED
+
+// System includes
+
+// Project includes
+#include "includes/serializer.h"
+
+// Application includes
+#include "custom_constitutive/bilinear_cohesive_3D_law.hpp"
+#include "poromechanics_application_variables.h"
+
+namespace Kratos
+{
+
+class KRATOS_API(POROMECHANICS_APPLICATION) SimplifiedBilinear3DLaw : public BilinearCohesive3DLaw
+{
+
+public:
+
+    /// Definition of the base class
+    typedef BilinearCohesive3DLaw BaseType;
+
+    KRATOS_CLASS_POINTER_DEFINITION(SimplifiedBilinear3DLaw);
+
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    // Default Constructor
+    SimplifiedBilinear3DLaw()
+    {
+    }
+
+    ConstitutiveLaw::Pointer Clone() const override
+    {
+        return Kratos::make_shared<SimplifiedBilinear3DLaw>(SimplifiedBilinear3DLaw(*this));
+    }
+
+    // Copy Constructor
+    SimplifiedBilinear3DLaw (const SimplifiedBilinear3DLaw& rOther) : BilinearCohesive3DLaw(rOther)
+    {
+    }
+
+    // Destructor
+    ~SimplifiedBilinear3DLaw() override
+    {
+    }
+
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    int Check(const Properties& rMaterialProperties, const GeometryType& rElementGeometry, const ProcessInfo& rCurrentProcessInfo) override;
+
+    void InitializeMaterial( const Properties& rMaterialProperties,const GeometryType& rElementGeometry,const Vector& rShapeFunctionsValues ) override;
+
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    void CalculateMaterialResponseCauchy (Parameters & rValues) override;
+
+    void FinalizeMaterialResponseCauchy (Parameters & rValues) override;
+
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+protected:
+
+protected:
+
+    struct ConstitutiveLawVariables
+    {
+        double YieldStress;
+        double YoungModulus;
+        double MaxTensileStress;
+        double MaxCompresiveStress;
+        double PoissonCoefficient;
+        double FrictionCoefficient;
+
+        Matrix CompressionMatrix;
+        Matrix WeightMatrix;
+
+        double EquivalentStrain;
+        bool LoadingFlag;
+        double LoadingFunction;
+    };
+
+    // Member Variables
+    double mStateVariable;
```

Comment:
Hi @joaquinirazabal ,
I would rather avoid creating an additional ConstitutiveLawVariables struct if possible. You can either add the variables that you need for the new simplified_bilinear_law inside the ConstitutiveLawVariables struct already defined in the bilinear_cohesive_3D_law.hpp, or rename the struct that you created in the simplified_bilinear_law.

Also, you can delete the member variable mStateVariable which is already defined in the base class.

Reply:
Thanks for your review, @ipouplana. I chose the option of adding the variables inside the ConstitutiveLawVariables struct already defined in the bilinear_cohesive_3D_law.hpp. I hope it is not an inconvenience for other Constitutive Laws.

If you think any of the new features of the new law can be applied to the previous bilinear cohesive law just tell me and I can implement them ;)